<a href="https://colab.research.google.com/github/vitorjensen/Python_para_Dados/blob/main/Projetos/An%C3%A1lise%20Explorat%C3%B3ria%20de%20Danos%20Informados/An%C3%A1lise_Explorat%C3%B3ria_de_Danos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##### Importando bibliotecas.

In [49]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings

##### **Objetivo:** Fazer uma análise exploratória de Danos em todos os estados e todo o período de 2024.

##### Os dados de Danos são todas as ocorrências de desastres e eventos de calamidade acontecidas nas regiões do Brasil e que envolveram mortes ou feridos.



##### Importando a base de dados para análise.

In [11]:
df = pd.read_csv("/content/drive/MyDrive/Danos_Informados.csv");

##### **Primeira parte da Análise Exploratória:**
Tratemento:
  - Remover colunas não utilizadas
  - Trocar os valores "0" por "Null"
  - Total de Nulos e Duplicados
  - Total de colunas e linhas

In [ ]:
# Visualizando o DataFrame
df.head()

In [ ]:
# Removendo colunas indesejadas
df.columns

index_coluna_dh_outros_afetados = df.columns.get_loc("DH_Outros Afetados") #Pegando a localização da coluna "DH_Outros Afetados"
colunas_para_deletar = df.columns[index_coluna_dh_outros_afetados + 1:] #Definindo o espaço de colunas a serem deletadas após.
df = df.drop(columns=colunas_para_deletar) # Comando Drop para deletar.
display(df.head())

In [48]:
# Trocando os valores de 0 por Null
df.replace(0, np.nan, inplace=True)
display(df)

,UF,Município,Registro,Protocolo,COBRADE,Status,População,DH_Mortos,DH_Feridos,DH_Enfermos,DH_Desabrigados,DH_Desalojados,DH_Desaparecidos,DH_Outros Afetados
0,SP,Caieiras,31/12/2024,SP-F-3509007-13214-20241230,13214 - Tempestade Local/Convectiva - Chuvas I...,Reconhecido,86623.0,1.0,NaN,NaN,4.0,4.0,NaN,16400.0
1,PE,Caruaru,02/01/2025,PE-F-2604106-14131-20241230,14131 - Incêndio Florestal - Incêndios em Parq...,Registro,314951.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,MG,Rio Preto,31/12/2024,MG-F-3155900-13214-20241230,13214 - Tempestade Local/Convectiva - Chuvas I...,Reconhecido,5292.0,NaN,NaN,NaN,NaN,37.0,NaN,2800.0
3,MG,Juiz de Fora,17/02/2025,MG-F-3136702-13214-20241230,13214 - Tempestade Local/Convectiva - Chuvas I...,Registro,517872.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,MG,Areado,09/01/2025,MG-F-3104304-13213-20241230,13213 - Tempestade Local/Convectiva - Granizo,Reconhecido,13729.0,NaN,NaN,NaN,NaN,60.0,NaN,985.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
6027,PR,Morretes,05/04/2024,PR-F-4116208-15110-20240101,15110 - Doenças infecciosas virais,Reconhecido,15718.0,NaN,NaN,254.0,NaN,NaN,NaN,761.0
6028,CE,Missão Velha,10/01/2024,CE-F-2308401-13214-20240101,13214 - Tempestade Local/Convectiva - Chuvas I...,Registro,34258.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN
6029,BA,Serra do Ramalho,18/01/2024,BA-F-2930154-14110-20240101,14110 - Estiagem,Registro,31646.0,NaN,NaN,NaN,NaN,NaN,NaN,1500.0
6030,RN,Porto do Mangue,01/01/2024,RN-F-2410256-11410-20240101,11410 - Erosão Costeira/Marinha,Registro,5217.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [26]:
# Somando os valores nulos.
df.isnull().sum().sum()

np.int64(36459)

In [28]:
# Identificando valores duplicados
df.duplicated().sum()

np.int64(0)

In [ ]:
df.info()

##### **Resumo do tratamento:**
  - Total de linhas da tabela: 6032
  - Total de colunas após a limpeza: 53 -> 14
  - Total de valores nulos na tabela: 36459
  - Total de valores duplicados: 0

##### **Iniciando segunda parte da Análise Exploratória:**
Definir:
  - Total de categorias distintas de Desastres.
  - Colunas Categóricas e Análise Univariada.
  - Colunas Númericas e Análise Univariada.
  - A Importância das colunas "DH".

In [ ]:
# Listando todas as categorias distintas de desastres
print(f"Categorias de desastres", end=" ")
print(df['COBRADE'].unique())

In [45]:
# Somando o total de categorias distintas de desastres
print(f"Categorias de desastres", end=" ")
print(df['COBRADE'].nunique())

Categorias de desastres 49


In [35]:
# Verificando colunas categóricas
colunas_categoricas = [coluna for coluna in df.columns
                       if df[coluna].dtype == 'object']


# Verificando colunas numéricas:
colunas_numericas = [coluna for coluna in df.columns
                     if df[coluna].dtype != 'object']
print(f"Temos um total de {len(colunas_categoricas)} colunas categóricas.")
print(f"Temos um total de {len(colunas_numericas)} colunas numéricas.")

Temos um total de 6 colunas categóricas.
Temos um total de 8 colunas numéricas.


In [ ]:
# Fazendo Análise Univariada de Atributos Categóricos
warnings.filterwarnings('ignore') #Ignorando avisos

plt.figure(figsize=(15, 15))
plt.suptitle("Análise Univariada de atributos categóricos", fontsize=20, fontweight='bold', alpha=0.8, y=1)
categorias = ['UF', 'Município', 'Registro', 'COBRADE', 'Status']

for i in range(len(categorias)):
  plt.subplot(2,2,i+1)
  sns.countplot(x=df[categorias[i]])
  plt.xlabel(categorias[i])
  plt.xticks(rotation=45)
  plt.tight_layout()


##### **Resumo da segunda parte da Análise Exploratória:**
- Categorias distintas de desastres: 49
- Colunas Categóricas: 8
- Colunas Numéricas: 6
- As colunas "DH's" (Danos Humanos) são importantes para que se possa entender a gravidade de um desastre para então classifica-lo ou a vulnerabilidade de um local.